Packs/Base/Scripts/GetIncidentsByQuery/GetIncidentsByQuery.py

```diff
@@ -2,244 +2,43 @@
-        parsed_incidents.append(new_incident)
-    return parsed_incidents
-
-
-def get_demisto_datetme_format(date_string):
-    if date_string:
-        date_object = None
-        # try to parse date string
-        try:
-            date_object = parser.parse(date_string)
-        except Exception:
-            pass
-        # try to parse relative time
-        if date_object is None and date_string.strip().endswith("ago"):
-            date_object = parse_relative_time(date_string)
-
-        if date_object:
-            return date_object.astimezone().isoformat('T')
-        else:
-            return None
-
-
-def get_incidents(query, time_field, size, from_date, to_date, fields_to_populate, include_context):
-    query_size = min(PAGE_SIZE, size)
-    args = {"query": query, "size": query_size, "sort": "%s.%s" % (time_field, "desc")}
-    # apply only when created time field
-    if time_field == 'created':
-        if from_date:
-            from_datetime = get_demisto_datetme_format(from_date)
-            if from_datetime:
-                args['fromdate'] = from_datetime
-            else:
-                demisto.results("did not set from date due to a wrong format: " + from_date)
-
-        if to_date:
-            to_datetime = get_demisto_datetme_format(to_date)
-            if to_datetime:
-                args['todate'] = to_datetime
-            else:
-                demisto.results("did not set to date due to a wrong format: " + from_date)
-
-    incident_list = []  # type: ignore
-    page = 0
-    while len(incident_list) < size:
-        incidents = get_incidents_by_page(args, page, fields_to_populate, include_context)
-        if not incidents:
-            break
-        incident_list += incidents
-        page += 1
-    return incident_list[:size]
-
-
-def get_comma_sep_list(value):
-    value = value.replace('|', ',')
-    return map(lambda x: x.strip(), value.split(","))
-
-
-def preprocess_incidents_fields_list(incidents_fields):
-    res = []
-    for field in incidents_fields:
-        field = field.strip()
-        for prefix in PREFIXES_TO_REMOVE:
-            if field.startswith(prefix):
-                field = field[len(prefix):]
-            res.append(field)
-    return res
+from GetIncidentsApiModule import *
+
+
+def encode_outputs(incidents: list[dict], output_format: str) -> str | bytes:
+    match output_format:
+        case "pickle":
+            return pickle.dumps(incidents, protocol=2)
+        case "json":
+            return json.dumps(incidents)
+        case _:
+            raise DemistoException(f"Invalid output format: {output_format}")
+
+
+def to_file_entry(incidents: list[dict], output_format: str) -> dict[str, Any]:
+    file_name = str(uuid.uuid4())
+    encoded_data = encode_outputs(incidents, output_format)
+    return fileResult(file_name, encoded_data) | {
+        "Contents": incidents,
+        "HumanReadable": f"Fetched {len(incidents)} incidents successfully",
```

Comment:
note that since `query` is calculated internally in the api module, the human readable of the script will not contain it anymore (instead I print it in the api module with `demisto.debug()`).